smoke-tests/apps/HttpPreaggregatedMetrics/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpPreaggregatedMetricsTestOtlpEnabled.java

```diff
@@ -0,0 +1,130 @@
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_21_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_23;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_23_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_8;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_8_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.WILDFLY_13_JAVA_8;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.WILDFLY_13_JAVA_8_OPENJ9;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.microsoft.applicationinsights.smoketest.schemav2.Data;
+import com.microsoft.applicationinsights.smoketest.schemav2.DataPoint;
+import com.microsoft.applicationinsights.smoketest.schemav2.Envelope;
+import com.microsoft.applicationinsights.smoketest.schemav2.MetricData;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@UseAgent
+abstract class HttpPreaggregatedMetricsTestOtlpEnabled {
+
+  @RegisterExtension
+  static final SmokeTestExtension testing = SmokeTestExtension.builder().useOtlpEndpoint().build();
+
+  @Test
+  @TargetUri("/httpUrlConnection")
+  void testMsSentToAmw() throws Exception {
+    verifyMetrics();
+  }
+
+  private void verifyMetrics() throws Exception {
+    List<Envelope> clientMetrics =
+        testing.mockedIngestion.waitForStandardMetricItems("dependencies/duration", 3);
+    List<Envelope> serverMetrics =
+        testing.mockedIngestion.waitForStandardMetricItems("requests/duration", 1);
+
+    verifyHttpClientPreAggregatedMetrics(clientMetrics);
+    verifyHttpServerPreAggregatedMetrics(serverMetrics);
+  }
+
+  private static void verifyHttpClientPreAggregatedMetrics(List<Envelope> metrics) {
+    assertThat(metrics.size()).isEqualTo(3);
+    // sort metrics based on result code
+    metrics.sort(
+        Comparator.comparing(
+            obj -> {
+              MetricData metricData = (MetricData) ((Data<?>) obj.getData()).getBaseData();
+              return metricData.getProperties().get("dependency/resultCode");
+            }));
+
+    // 1st pre-aggregated metric
+    Envelope envelope1 = metrics.get(0);
+    MetricData md1 = (MetricData) ((Data<?>) envelope1.getData()).getBaseData();
+    validateMetricData(md1);
+
+    // 2nd pre-aggregated metric
+    Envelope envelope2 = metrics.get(1);
+    MetricData md2 = (MetricData) ((Data<?>) envelope2.getData()).getBaseData();
+    validateMetricData(md2);
+
+    // 3rd pre-aggregated metric
+    Envelope envelope3 = metrics.get(2);
+    MetricData md3 = (MetricData) ((Data<?>) envelope3.getData()).getBaseData();
+    validateMetricData(md3);
```

Comment:
is sorting needed? looks like same verification is performed for all 3


```suggestion
    for (Envelope envelope : metrics) {
      MetricData md = (MetricData) ((Data<?>) envelope.getData()).getBaseData();
      validateMetricData(md);
    }
```